pykokkos/interface/type_inference.py

```diff
@@ -0,0 +1,279 @@
+import inspect
+from dataclasses import dataclass
```

Comment:
should this file be args_type_inference?

Reply:
I think I will refactor this once I know how to deal with workunit bodies. It might be that we can expand/use the same file